### PR TITLE
Added Torch Pruning for Chipnet and Slimming and removed wanted files from models folder.

### LINKED ISSUE #99: Fix the models folder structure in the trailmet library.
Currently, there are multiple definitions of the same architecture in different files. Remove the unwanted files and fix the imports accordingly.

### PR DESCRIPTION
**Pull request by animesh-007**
- [x] Chipnet and Slimming pruning is added using Torch Pruning
- [x] Removed the `models` folder from `trailmet/models` folder. Partially fixes #99

### CODE REVIEW

File: trailmet/algorithms/prune/chipnet.py
Comment (Arnav0400) [inline trailmet/algorithms/prune/chipnet.py:1]: why are we removing seed function here?

File: trailmet/algorithms/prune/network_slimming.py [hunk 1]
```diff
@@ -96,14 +73,13 @@ def __init__(self,**kwargs):
...
         self.lr = kwargs.get('learning_rate',2e-3)
-        #self.save_dir = kwargs.get('save_dir')
         self.prune_schema = os.path.join(kwargs.get('schema_root') , f"schema/{self.net}.json")
         self.sparsity_train = kwargs.get('sparsity_train',True)
         self.fine_tune_epochs = kwargs.get('fine_tune_epochs',165)
         self.fine_tune_lr = kwargs.get('fine_tune_learning_rate',1e-4)
         self.prune_ratio = kwargs.get('prune_ratio',0.5)
-#        self.bn_weight_vis_period = kwargs.get('bn_weight_vis_period')
         self.log_base = f"{self.net}_{self.dataset}.pth"
+        self.model = Model(self.net,self.num_classes)
```
Comment (Arnav0400) [inline trailmet/algorithms/prune/network_slimming.py]: Why are we using a custom Model class here, cant we use the generic one ?

File: trailmet/algorithms/prune/chipnet.py
Comment (animesh-007) [reply trailmet/algorithms/prune/chipnet.py:1]: Initially, it was causing a circular import error. I have found a workaround to fix it.

File: trailmet/algorithms/prune/network_slimming.py (hunk 1 above)
Comment (animesh-007) [reply trailmet/algorithms/prune/network_slimming.py]: I also agree that the custom model class can be removed, and a generic one can be used. I have used the generic model class for the torch pruning code. Let me know if the same fix should be made in the network slimming original code.

**Review by Arnav0400 — APPROVED** [review]
